File: C_Sharp/Framework/LazyLoad/LazyLoader.cs
namespace Framework.LazyLoad
{
    public interface ILazyLoader<TEntity, TSearchInput>
    {
        void LoadAllFields(ref TEntity entity, TSearchInput criteria);
    }

    public abstract class LazyLoader<TEntity, TSearchInput> : ILazyLoader<TEntity, TSearchInput>
        where TEntity : LazyLoadDomainObject<TSearchInput>
    {
        public abstract void LoadAllFields(ref TEntity entity, TSearchInput criteria);
    }
}

File: Android/framework/src/main/java/com/LazyLoad/LazyLoader.java
package com.LazyLoad;

import com.Interfaces.ILazyLoader;

/**
 * Created by aiko on 4/2/17.
 */

public abstract class LazyLoader <TEntity extends LazyLoadDomainObject<TSearchInput>, TSearchInput> implements ILazyLoader<TEntity, TSearchInput> {
    public abstract void LoadAllFields(TEntity entity, TSearchInput criteria);
}
